pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -173,18 +174,24 @@ func (c *Controller) Services() []*model.Service {
 			services = append(services, svcs...)
 		} else {
 			for _, s := range svcs {
-				sp, ok := smap[s.Hostname]
+				previous, ok := smap[s.Hostname]
 				if !ok {
 					// First time we see a service. The result will have a single service per hostname
 					// The first cluster will be listed first, so the services in the primary cluster
 					// will be used for default settings. If a service appears in multiple clusters,
 					// the order is less clear.
-					smap[s.Hostname] = s
+					smap[s.Hostname] = index
+					index++
 					services = append(services, s)
 				} else {
+					// If previous service has not been deep copied
+					if len(services[previous].ClusterVIPs.GetAddresses()) < 2 {
```

Comment:
can please explain in comments why we are checking < 2?

Reply:
OK, this is a optimization. To make deepcopy at most once for like N clusters. I will add comments